app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -98,8 +100,16 @@ void afterGroupNumChanged() {
     // Given the number of names per group or number of groups, returns the corresponding number to use all the names
     private String getMatchingNumber(String inputtedNumber) {
         int newNumber = Integer.parseInt(inputtedNumber);
-        int offset = (int) Math.ceil((double) settings.getNameListSize() / newNumber);
-        return String.valueOf(offset);
+        double offset = (double) settings.getNameListSize() / newNumber;
+        checkGroupBalance(offset);
+        int matchingNumber = (int) Math.ceil(offset);
+        return String.valueOf(matchingNumber);
+    }
+
+    // check if the groups have the same names quantity
+    private void checkGroupBalance(double ratio) {
```

Comment:
We should change this method to something like maybeShowUnevenGroupsWarning(). It can extract the numbers from the views itself and then do the math. Make sure that it doesn't do anything if one of the values is empty.